### PR TITLE
Fix missing version links and dates in CHANGELOG

### PR DESCRIPTION
**Pull request by sunsided**
Summary of the changes (Less than 80 chars)

- Add missing release links and dates to [CHANGELOG.md](CHANGELOG.md)

I have noticed the changelog isn't actively maintained anymore with the release descriptions being available on the Releases page(s) instead.

However, the document used to follow [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) but then deviated after the switch from 0.8.x to 9.0.0. I added the missing links and dates in so that it's compliant again.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -883,7 +883,42 @@ Return argument value as field definition runtimeType if requested as object typ
 - Diagnostic source which can be used to track field execution times and other events.
 - Implementing a directive middleware has now become much easier with this release. We have built the authorize-directive with these new APIs.
 
-[unreleased]: https://github.com/ChilliCream/graphql-platform/compare/0.8.2...HEAD
+[unreleased]: https://github.com/ChilliCream/graphql-platform/compare/12.0.1...HEAD
+[12.0.1]: https://github.com/ChilliCream/graphql-platform/compare/12.0.0...12.0.1
+[12.0.0]: https://github.com/ChilliCream/graphql-platform/compare/11.1.0...12.0.0
+[11.1.0]: https://github.com/ChilliCream/graphql-platform/compare/10.0.9...11.1.0
```
Comment (PascalSenn) [inline CHANGELOG.md:889]: Are these links correct? comparing v10 with v11?

Comment (sunsided) [reply CHANGELOG.md:889]: Some of them did seem odd to me too, but this is the pattern used by the original CHANGELOG prior to 9.0.0, comparing the current release with the one immediately preceding it:

```markdown
[0.8.2]: https://github.com/ChilliCream/graphql-platform/compare/0.8.1...0.8.2
[0.8.1]: https://github.com/ChilliCream/graphql-platform/compare/0.8.0...0.8.1
[0.8.0]: https://github.com/ChilliCream/graphql-platform/compare/0.7.0...0.8.0
[0.7.0]: https://github.com/ChilliCream/graphql-platform/compare/0.6.11...0.7.0
[0.6.11]: https://github.com/ChilliCream/graphql-platform/compare/0.6.10...0.6.11
```

I am aware there is a host of other releases in between that are simply not listed in this document, but I didn't want to go into back-porting them not knowing if this document even has a future. :)

Comment (sunsided) [reply CHANGELOG.md:889]: That said, you are right, this is indeed a typo. Fixing it now.

Comment (sunsided) [reply CHANGELOG.md]: Should be good now.

**Review by glen-84 — CHANGES_REQUESTED** [review]

**Comment by glen-84** [comment]
@michaelstaib

This can be merged after applying the suggestion, but I think that this file should be deleted afterwards, since it's no longer being kept up-to-date. I can send a PR if you want.

**Review by michaelstaib — APPROVED** [review]

**Pull request merged**